libgit/repo.go

```diff
@@ -545,3 +571,167 @@ func DeleteRepo(
 		ctx, repoNode, normalizedRepoName, deletedReposNode,
 		normalizedRepoName+dirSuffix)
 }
+
+func renameRepoInConfigFile(
+	ctx context.Context, repoFS billy.Filesystem, newRepoName string) error {
+	// Assume lock file is already taken for both the old repo and the
+	// new one.
+	f, err := repoFS.OpenFile(kbfsConfigName, os.O_RDWR, 0600)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	buf, err := ioutil.ReadAll(f)
+	if err != nil {
+		return err
+	}
+	c, err := configFromBytes(buf)
+	if err != nil {
+		return err
+	}
+	c.Name = newRepoName
+	buf, err = c.toBytes()
+	if err != nil {
+		return err
+	}
+	_, err = f.Seek(0, io.SeekStart)
+	if err != nil {
+		return err
+	}
+	err = f.Truncate(0)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(buf)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// RenameRepo renames the repo from an old name to a new name.  It
+// leaves a symlink behind so that old remotes will continue to work.
+// The caller is responsible for syncing the FS and flushing the
+// journal, if desired.
+func RenameRepo(
+	ctx context.Context, config libkbfs.Config, tlfHandle *libkbfs.TlfHandle,
+	oldRepoName, newRepoName string) error {
+	kbfsOps := config.KBFSOps()
+	rootNode, _, err := kbfsOps.GetOrCreateRootNode(
+		ctx, tlfHandle, libkbfs.MasterBranch)
+	if err != nil {
+		return err
+	}
+	normalizedOldRepoName := normalizeRepoName(oldRepoName)
+	normalizedNewRepoName := normalizeRepoName(newRepoName)
+
+	repoNode, _, err := kbfsOps.Lookup(ctx, rootNode, kbfsRepoDir)
+	if err != nil {
+		return err
+	}
+
+	// Does the old repo definitely exist?
+	_, _, err = kbfsOps.Lookup(ctx, repoNode, normalizedOldRepoName)
+	if err != nil {
+		return err
+	}
+
+	if oldRepoName == newRepoName {
+		// The names are the same, nothing else to do.
+		return nil
+	}
+
+	fs, err := libfs.NewFS(ctx, config, tlfHandle, path.Join(kbfsRepoDir), "")
+	if err != nil {
+		return err
+	}
+
+	oldRepoFS, err := fs.Chroot(normalizedOldRepoName)
+	if err != nil {
+		return err
+	}
+
+	// Take locks in both repos during rename (same lock that's taken
+	// for new repo creation).
+	oldLockFile, err := takeConfigLock(
+		oldRepoFS.(*libfs.FS), tlfHandle, oldRepoName)
+	if err != nil {
+		return err
+	}
+	defer oldLockFile.Close()
+
+	if normalizedOldRepoName == normalizedNewRepoName {
```

Comment:
So this case is for case changes, and `.git` suffix addition/removal?

Related question: should [`normalizeRepoName`](https://github.com/keybase/kbfs/blob/4ed4da1f519db79f8edbce19500a02a7bd3a7327/libgit/repo.go#L295) do the `strings.TrimSuffix` in a loop until the string doesn't change, in case something like `repo.git.git` was passed in?

Reply:
> So this case is for case changes, and .git suffix addition/removal?

Yes.

> Related question: should normalizeRepoName do the strings.TrimSuffix in a loop until the string doesn't change, in case something like repo.git.git was passed in?

Whoa, I didn't realize github did this, but you're right.  I guess it should (in a separate PR).

Reply:
> I didn't realize github did this

Hmm I didn't know that!